sdks/java/transform-service/launcher/src/main/java/org/apache/beam/sdk/transformservice/launcher/TransformServiceLauncher.java

```diff
@@ -0,0 +1,285 @@
+    runDockerComposeCommand("up");
+  }
+
+  public synchronized void shutdown() throws IOException {
+    runDockerComposeCommand("down");
+  }
+
+  public synchronized void status() throws IOException {
+    runDockerComposeCommand("ps");
+  }
+
+  public synchronized void waitTillUp(int timeout) throws IOException, TimeoutException {
+    timeout = timeout <= 0 ? DEFAULT_START_WAIT_TIME : timeout;
+    String statusFileName = getStatus();
+
+    long startTime = System.currentTimeMillis();
+    while (System.currentTimeMillis() - startTime < timeout) {
+      try {
+        // We are just waiting for a local process. No need for exponential backoff.
+        this.wait(1000);
+      } catch (InterruptedException e) {
+        // Ignore and retry.
+      }
+
+      String output = String.join(" ", java.nio.file.Files.readAllLines(Paths.get(statusFileName)));
+      if (!output.isEmpty()) {
+        if (output.contains("transform-service")) {
+          // Transform service was started since we found matching logs.
+          return;
+        }
+      }
+    }
+
+    throw new TimeoutException(
+        "Transform Service did not start in " + timeout / 1000 + " seconds.");
+  }
+
+  private synchronized String getStatus() throws IOException {
+    File outputOverride = File.createTempFile("output_override", null);
+    runDockerComposeCommand("ps", outputOverride);
+
+    return outputOverride.getAbsolutePath();
+  }
+
+  public static void main(String[] args) throws IOException, TimeoutException {
+    String projectName = null;
+    String command = null;
+    int port = -1;
+    String beamVersion = null;
+
+    if (args.length % 2 != 0) {
+      throw new IllegalArgumentException(
+          "Incorrect number of arguments. Supported arguments are " + COMMAND_POSSIBLE_VALUES);
+    }
+
+    Iterator<String> argIter = Arrays.stream(args).iterator();
+    while (argIter.hasNext()) {
+      String key = argIter.next();
+      String value = argIter.next();
+
+      if (key.equalsIgnoreCase("--project_name")) {
+        projectName = value;
+      } else if (key.equalsIgnoreCase("--port")) {
+        port = Integer.parseInt(value);
+      } else if (key.equalsIgnoreCase("--command")) {
+        command = value;
+      } else if (key.equalsIgnoreCase("--beam_version")) {
+        beamVersion = value;
+      }
+    }
+
+    if (command == null) {
+      throw new IllegalArgumentException(
+          "\"command\" argument must be specified, Valid values are " + COMMAND_POSSIBLE_VALUES);
+    }
+
+    System.out.println("===================================================");
+    System.out.println(
+        "Starting the Beam Transform Service at "
+            + (port < 0 ? "the default port." : ("port " + Integer.toString(port) + ".")));
+    System.out.println("===================================================");
+
+    TransformServiceLauncher service = TransformServiceLauncher.forProject(projectName, port);
+    if (beamVersion != null) {
```

Comment:
does this mean we allow different Java and Python beam versions? If so, is it possible to get the Java version and if needed, warn user about potential version mismatch?

Reply:
No, we use the same version for both Java and Python (currently).

Reply:
I played around with the launcher. If launching with
`./gradlew :sdks:java:transform-service:launcher:run --args="--command up --port 5005 --beam_version 2.47.0"`

I do see a log "Pulling transform-service (apache/beam_transform_service_controller:2.47.0)..."

If launching with
`./gradlew :sdks:java:transform-service:launcher:run --args="--command up --port 5005"` (without beam_version),

then the log shows 
`no such image: apache/beam_transform_service_controller:$BEAM_VERSION: invalid reference format`

which means the docker compose variable "$BEAM_VERSION" is not assigned when this parameter not provided. For convenience we should still have a default value that is the same as the SDK version. Otherwise this parameter is effectively mandatory

Reply:
Setting a default version requires a dependency to Beam core module which I'd like to avoid (due to the increased size of the shaded jar).

Instead I made beam_version a required parameter (for the command line).

I already set the default Beam version when this utility is used by Beam via PythonExternalTransform.

Reply:
Thanks, understand. Would you mind putting a comment about the intentially light weight launcher and avoid beam core dependency in shaded jar in the build.gradle?

Reply:
Also, may worth announce in CHANGE.md for this new feature. could be done as follow up

Reply:
Yeah, will add seperately.